src/plugins/dashboard/public/application/actions/clone_panel_action.tsx

```diff
@@ -109,41 +110,64 @@ export class ClonePanelAction implements Action<ClonePanelActionContext> {
       : baseTitle + ` (${clonedTag} ${similarBaseTitlesCount})`;
   }
 
+  private async addCloneToLibrary(
+    objectIdToClone: string,
+    embeddableType: string
+  ): Promise<string> {
+    const savedObjectToClone = await this.core.savedObjects.client.get<SavedObject>(
+      embeddableType,
+      objectIdToClone
+    );
+
+    // Clone the saved object
+    const newTitle = await this.getUniqueTitle(savedObjectToClone.attributes.title, embeddableType);
+    const clonedSavedObject = await this.core.savedObjects.client.create(
+      embeddableType,
+      {
+        ..._.cloneDeep(savedObjectToClone.attributes),
+        title: newTitle,
+      },
+      { references: _.cloneDeep(savedObjectToClone.references) }
+    );
+    return clonedSavedObject.id;
+  }
```

Comment:
This logic is *only* necessary for `type='search'` panels since they can never be `byValue` - they must *always* be `byReference`. Lots of code for one very specific case but 🤷

Reply:
Someday we should make `search` panels support by value! Then we could likely remove all this library logic and just clone the raw embeddable input for embeddables that cannot do `getInputAsValueType`. That would simplify this _greatly_